Условные операторы/Условные операторы/Program.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+namespace If
+{
+    class Program
+    {
+        static int FunctionResult(int x)
+        {
+            if (x < -4)
+                return 46;
+            else if ((x >= -4) && (x <= 2))
+                return 3 * x * x - 2;
+            else
+                return 10;
+        }
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите значение x");
+            int x = int.Parse(Console.ReadLine());
+            int result = FunctionResult(x);
+            Console.WriteLine("Результат функции f(x) - " + result);
+            Console.ReadKey();
+        }
+    }
```

Comment:
1. В задании не сказано, что аргумент целочисленный. Нужно использовать числовой тип с плавающей точкой.  Это ошибка.
2. Размещайте методы в классе в порядке сначала вызывающий, потом вызываемый и разделяйте их код пустой строкой. Так программа читается легче.
3. Учитывайте приоритеты операций. Это позволит избежать лишних скобок.
4. Избегайте лишних проверок.
5. Если внутри if стоит return, то дальнейший else излишен.
```suggestion
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Введите значение x");
            double x = double.Parse(Console.ReadLine());
            double result = FunctionResult(x);
            Console.WriteLine("Результат функции f(x) - " + result);
            Console.ReadKey();
        }
        
        static double FunctionResult(double x)
        {
            if (x < -4)
                return 46;
            
            if (x <= 2)
                return 3 * x * x - 2;

            return 10;    
        }
    }
```